File: route-service/utils/compose_name_helper.py
from typing import List

from dto.mapbox.response.mapbox_direction_response import MapboxWayPoint


class ComposeNameHelper:
    @staticmethod
    def compose_route_name(waypoints: List[MapboxWayPoint]) -> str:
        top_names = [w.name for w in waypoints[:min(3, len(waypoints))]]
        return " - ".join(top_names)

    @staticmethod
    def compose_location_name(ward: str, district: str, city: str) -> str:
        return ', '.join(filter(None, [ward, district, city]))

File: core-service/src/main/java/com/stride/tracking/coreservice/utils/ComposeNameHelper.java
package com.stride.tracking.coreservice.utils;

import com.stride.tracking.dto.mapbox.response.MapboxWayPoint;

import java.util.List;
import java.util.stream.Collectors;
import java.util.stream.Stream;

public class ComposeNameHelper {
    private ComposeNameHelper() {
    }

    public static String composeRouteName(List<MapboxWayPoint> waypoints) {
        if (waypoints == null || waypoints.isEmpty()) {
            return "";
        }

        return waypoints.stream()
                .limit(3)
                .map(MapboxWayPoint::getName)
                .collect(Collectors.joining(" - "));
    }

    public static String composeLocationName(String ward, String district, String city) {
        return Stream.of(ward, district, city)
                .filter(part -> part != null && !part.isBlank())
                .collect(Collectors.joining(", "));
    }
}
